test/newton.jl

```diff
@@ -0,0 +1,57 @@
+# Newton's method for finding roots of a function
+# wrapped as a very simple iterative algorithm
+
+using AlgorithmsInterface
+import AlgorithmsInterface: initialize_state, initialize_state!, is_finished, solve!, step!
+using Test
+
+# Defining the structs
+# ------------------
+struct RootFindingProblem <: Problem
+    f::Function
+    df::Function
+end
+
+struct NewtonMethod{S} <: Algorithm
+    stopping_criterion::S
```

Comment:
This is one thing I feared. I would like to find a way to not “double” stuff. The SC is now double here.

Can we somehow avoid such a duplication? The SC has to be in the State, since it is something that changes during the iterations, so maybe it should not be here? Or something else should be here?

Reply:
Give me one second, I'm typing up some of my thoughs :)

Reply:
But besides this point, the implementation is roughly what I had in mind, besides maybe a bit of syntactic sugar here and there....

So my main problem would probably be: How to properly split between “Static Algorithm stuff” and “changing Algorithm state stuff”